src/Microsoft.NetCore.Analyzers/UnitTests/Performance/UsePropertyInsteadOfCountMethodWhenAvailableTests.cs

```diff
@@ -0,0 +1,452 @@
+    public static int? M() => GetData()?.Count();
+}}
+",
+                VerifyCS.Diagnostic(UsePropertyInsteadOfCountMethodWhenAvailableAnalyzer.RuleId)
+                    .WithLocation(6, 41)
+                    .WithArguments(propertyName),
+                $@"using System;
+using System.Linq;
+public static class C
+{{
+    public static {type} GetData() => default;
+    public static int? M() => GetData()?.{propertyName};
+}}
+");
+
+        [Theory]
+        [InlineData("string()", nameof(Array.Length))]
+        [InlineData("System.Collections.Immutable.ImmutableArray(Of Integer)", nameof(ImmutableArray<int>.Length))]
+        public static Task Basic_Fixed(string type, string propertyName)
+            => VerifyVB.VerifyCodeFixAsync(
+                $@"Imports System
+Imports System.Linq
+Public Module M
+    Public Function GetData() As {type}
+        Return Nothing
+    End Function
+    Public Function F() As Integer
+        Return GetData().Count()
+    End Function
+End Module
+",
+                VerifyCS.Diagnostic(UsePropertyInsteadOfCountMethodWhenAvailableAnalyzer.RuleId)
+                    .WithLocation(8, 16)
+                    .WithArguments(propertyName),
+                $@"Imports System
+Imports System.Linq
+Public Module M
+    Public Function GetData() As {type}
+        Return Nothing
+    End Function
+    Public Function F() As Integer
+        Return GetData().{propertyName}
+    End Function
+End Module
+");
+
+        [Theory]
+        [InlineData("string()", nameof(Array.Length))]
+        [InlineData("System.Collections.Immutable.ImmutableArray(Of Integer)?", nameof(ImmutableArray<int>.Length))]
+        public static Task Basic_Conditional_Fixed(string type, string propertyName)
+            => VerifyVB.VerifyCodeFixAsync(
+                $@"Imports System
+Imports System.Linq
+Public Module M
+    Public Function GetData() As {type}
+        Return Nothing
+    End Function
+    Public Function F() As Integer
+        Return GetData()?.Count()
+    End Function
+End Module
+",
+                VerifyCS.Diagnostic(UsePropertyInsteadOfCountMethodWhenAvailableAnalyzer.RuleId)
+                    .WithLocation(8, 26)
+                    .WithArguments(propertyName),
+                $@"Imports System
+Imports System.Linq
+Public Module M
+    Public Function GetData() As {type}
+        Return Nothing
+    End Function
+    Public Function F() As Integer
+        Return GetData()?.{propertyName}
+    End Function
+End Module
+");
+
+        [Theory]
+        [InlineData("System.Collections.Generic.IEnumerable<int>")]
+        public static Task CSharp_NoDiagnostic(string type)
+            => VerifyCS.VerifyAnalyzerAsync(
+                $@"using System;
+using System.Linq;
+public static class C
+{{
+    public static {type} GetData() => default;
+    public static int M() => GetData().Count();
+}}
+");
+
+        [Theory]
+        [InlineData("System.Collections.Generic.List(Of Integer)")]
```

Comment:
Shouldn't we be flagging this case? and the IList and ICollection cases below?

Reply:
Maybe I'm too restrictive, but I'm making sure the only the interface implementations are flagged.